### PR TITLE
added publish Github Action

### PR DESCRIPTION
**Pull request by sabasabo**
### Issue & Steps to Reproduce / Feature Request
Publishing a new release process is too manual.

### Solution
Using GitHub release mechanism so we can have an easy way to publish a new release.

### CODE REVIEW

File: .github/workflows/auto_assign.yml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-# used by ./workflows/pr-auto-assign.yml
-addAssignees: author
```
Comment (eranelbaz) [inline .github/workflows/auto_assign.yml:2]: why did you remove this?

File: .github/workflows/publish-npm.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - uses: actions/checkout@v2
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm ci
+      - run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (eranelbaz) [inline .github/workflows/publish-npm.yml:23]: extra line plz

File: .github/workflows/auto_assign.yml (hunk 1 above)
Comment (sabasabo) [reply .github/workflows/auto_assign.yml:2]: It was duplicated from .github/auto_assign.yml and Github showed errors.

In the env0 repo, this file sits in .github/ and not in ...workflows

File: .github/workflows/publish-npm.yml (hunk 2 above)
Comment (sabasabo) [reply .github/workflows/publish-npm.yml:23]: done: e29a5deec6b0b7c83c69245269a30d2202e4eb2c

**Review by eranelbaz — APPROVED** [review]
